sys/net/netif/netif.c

```diff
@@ -56,7 +56,7 @@ __attribute__((weak)) int16_t netif_get_id(const netif_t *netif)
     return -1;
 }
 
-netif_t *netif_get_by_name(const char *name)
+netif_t *netif_get_by_name_buffer(const char *name, size_t name_len)
 {
     assert(name);
     list_node_t *node = netif_list.next;
```

Comment:
In my version I added a check here to prevent exceeding `CONFIG_NETIF_NAMELENMAX`:

```suggestion
    if (name_len > CONFIG_NETIF_NAMELENMAX) {
        return NULL;
    }
    list_node_t *node = netif_list.next;
```